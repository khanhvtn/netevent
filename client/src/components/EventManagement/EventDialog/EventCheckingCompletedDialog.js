import React from 'react';
import {
    CircularProgress,
    Button,
    Dialog,
    DialogTitle,
    DialogContent,
    DialogActions,
    DialogContentText,
    Slide
} from '@material-ui/core';
import { useSelector } from 'react-redux';
//import useStyles in the last
import useStyles from './styles';

const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="up" ref={ref} {...props} />;
});

const EventCheckingCompletedDialog = ({
    openCheckingCompletedDialog,
    handleToggleDialogCheckingCompleted,
    handleUpdateEventStatus
}) => {
    const css = useStyles();

    const { isLoading } = useSelector((state) => ({
        isLoading: state.event.isLoading
    }));

    return (
        <>
            {/* Dialog Checking completed */}
            <Dialog
                TransitionComponent={Transition}
                open={openCheckingCompletedDialog}
                onClose={handleToggleDialogCheckingCompleted}
                aria-labelledby="delete-dialog"
                fullWidth
                className={css.dialogDeleteFac}
                aria-describedby="delete-dialog-description">
                <DialogTitle id="delete-dialog">
                    {'Confirm completed event!'}
                </DialogTitle>
                <DialogContent>
                    <DialogContentText id="delete-dialog-description">
                        Your event currently out of date! You should update your
                        event's status.
                    </DialogContentText>
                </DialogContent>
                <DialogActions className={css.dialogCheckingActions}>
                    <Button
                        fullWidth
                        disabled={isLoading ? true : false}
                        onClick={handleUpdateEventStatus}
                        variant="contained"
                        color="primary">
                        {isLoading ? (
                            <CircularProgress size={25} color="inherit" />
                        ) : (
                            'Confirm Completed'
                        )}
                    </Button>
                </DialogActions>
            </Dialog>
        </>
    );
};

export default EventCheckingCompletedDialog;
