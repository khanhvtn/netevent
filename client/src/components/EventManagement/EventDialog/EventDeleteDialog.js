import React from 'react';
import {
    CircularProgress,
    Button,
    Dialog,
    DialogTitle,
    DialogContent,
    DialogActions,
    DialogContentText,
    Slide,
} from '@material-ui/core';
import { useSelector } from 'react-redux';
//import useStyles in the last
import useStyles from './styles';

const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="up" ref={ref} {...props} />;
});

const EventDeleteDialog = ({
    openDeleteDialog,
    handleToggleDialogDelete,
    handleDeleteEvent,
    isRecoveryMode,
}) => {
    const css = useStyles();

    const { isLoading } = useSelector((state) => ({
        isLoading: state.event.isLoading,
    }));

    return (
        <>
            {/* Dialog Confirm Delete */}
            <Dialog
                TransitionComponent={Transition}
                open={openDeleteDialog}
                onClose={handleToggleDialogDelete}
                aria-labelledby="delete-dialog"
                fullWidth
                className={css.dialogDeleteFac}
                aria-describedby="delete-dialog-description"
            >
                <DialogTitle id="delete-dialog">{'Warning!!!'}</DialogTitle>
                <DialogContent>
                    <DialogContentText id="delete-dialog-description">
                        {isRecoveryMode
                            ? 'This action will delete the event permanently. Are you sure with your action ?'
                            : 'Are you sure with your action ?'}
                    </DialogContentText>
                </DialogContent>
                <DialogActions className={css.dialogActions}>
                    <Button
                        disabled={isLoading ? true : false}
                        onClick={handleToggleDialogDelete}
                        color="default"
                    >
                        Cancel
                    </Button>
                    <Button
                        disabled={isLoading ? true : false}
                        variant="contained"
                        onClick={handleDeleteEvent}
                        color="secondary"
                    >
                        {isLoading ? (
                            <CircularProgress size={25} color="inherit" />
                        ) : (
                            'Delete'
                        )}
                    </Button>
                </DialogActions>
            </Dialog>
        </>
    );
};

export default EventDeleteDialog;
