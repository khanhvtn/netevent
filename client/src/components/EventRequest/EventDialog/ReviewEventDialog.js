import React, { useEffect } from 'react';
import {
  CircularProgress,
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  DialogContentText,
  Slide
} from '@material-ui/core';
import { useSelector } from 'react-redux';
//import useStyles in the last
import useStyles from './styles';

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

const ReviewEventDialog = ({
  openReviewEventDialog,
  handleToggleDialogReviewEvent,
  handleUpdateEventStatus
}) => {
  const css = useStyles();

  const { isLoading, updateEventSuccess } = useSelector((state) => ({
    isLoading: state.event.isLoading,
    updateEventSuccess: state.event.updateSuccess
  }));

  //useEffect for create event success
  useEffect(() => {
    if (updateEventSuccess) {
      handleToggleDialogReviewEvent();
    }
  }, [updateEventSuccess]);

  return (
    <>
      {/* Dialog Review Event Status */}
      <Dialog
        TransitionComponent={Transition}
        open={openReviewEventDialog}
        onClose={handleToggleDialogReviewEvent}
        aria-labelledby="review-event-dialog"
        fullWidth
        className={css.dialogDeleteFac}
        aria-describedby="review-event-dialog-description">
        <DialogTitle id="review-event-dialog">
          {'Confirm completed event!'}
        </DialogTitle>
        <DialogContent>
          <DialogContentText id="review-event-dialog-description">
            Process the event status.
          </DialogContentText>
        </DialogContent>
        <DialogActions className={css.dialogCheckingActions}>
          <Button
            fullWidth
            disabled={isLoading ? true : false}
            onClick={() => handleUpdateEventStatus(true, 'approve')}
            variant="contained"
            color="primary">
            {isLoading ? (
              <CircularProgress size={25} color="inherit" />
            ) : (
              'Approve'
            )}
          </Button>
          <Button
            fullWidth
            disabled={isLoading ? true : false}
            onClick={() => handleUpdateEventStatus(false, 'approve')}
            variant="contained"
            color="secondary">
            {isLoading ? (
              <CircularProgress size={25} color="inherit" />
            ) : (
              'Reject'
            )}
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default ReviewEventDialog;
