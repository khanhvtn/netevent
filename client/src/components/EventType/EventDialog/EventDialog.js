import React from 'react';
import {
    CircularProgress,
    Button,
    Dialog,
    DialogTitle,
    DialogContent,
    DialogActions,
    TextField,
    DialogContentText,
    Slide
} from '@material-ui/core';
//import useStyles in the last
import useStyles from './styles';
const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="up" ref={ref} {...props} />;
});
const EventDialog = ({
    openCreateAndUpdateDialog,
    handleToggleDialogCreateAndUpdate,
    isCreateMode,
    handleChange,
    name,
    openDeleteDialog,
    handleCreateAndUpdate,
    handleToggleDialogDelete,
    handleDelete,
    isLoading,
    errors,
    createSuccess,
    isRecoveryMode
}) => {
    const css = useStyles();
    return (
        <div>
            {/* Dialog Create and Update */}
            <Dialog
                TransitionComponent={Transition}
                maxWidth="sm"
                open={openCreateAndUpdateDialog}
                onClose={(e) => handleToggleDialogCreateAndUpdate(e)}
                aria-labelledby="form-dialog-title">
                <DialogTitle id="form-dialog-title">
                    {isCreateMode
                        ? 'Create New Event Type'
                        : 'Update a Event Type'}
                </DialogTitle>
                <DialogContent>
                    <TextField
                        disabled={isLoading || createSuccess ? true : false}
                        className={css.textField}
                        helperText={errors?.name ? errors?.name : ''}
                        error={errors?.name ? true : false}
                        variant="outlined"
                        onChange={handleChange}
                        id="name"
                        value={name}
                        name="name"
                        label="Name"
                        type="text"
                        fullWidth
                    />
                </DialogContent>
                <DialogActions className={css.dialogActions}>
                    <Button
                        disabled={isLoading || createSuccess ? true : false}
                        variant="contained"
                        onClick={handleToggleDialogCreateAndUpdate}
                        color="default">
                        Cancel
                    </Button>
                    <Button
                        disabled={isLoading || createSuccess ? true : false}
                        variant="contained"
                        onClick={handleCreateAndUpdate}
                        color="primary">
                        {isLoading ? (
                            <CircularProgress size={25} color="inherit" />
                        ) : isCreateMode ? (
                            'Create'
                        ) : (
                            'Update'
                        )}
                    </Button>
                </DialogActions>
            </Dialog>

            {/* Dialog Confirm Delete */}
            <Dialog
                maxWidth="sm"
                fullWidth
                TransitionComponent={Transition}
                open={openDeleteDialog}
                onClose={handleToggleDialogDelete}
                aria-labelledby="delete-dialog"
                aria-describedby="delete-dialog-description">
                <DialogTitle id="delete-dialog">{'Warning!!!'}</DialogTitle>
                <DialogContent>
                    <DialogContentText id="delete-dialog-description">
                        {isRecoveryMode
                            ? 'This action will delete the event type permanently. Are you sure with your action ?'
                            : 'Are you sure with your action ?'}
                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button
                        disabled={isLoading ? true : false}
                        variant="contained"
                        onClick={handleDelete}
                        color="secondary">
                        {isLoading ? (
                            <CircularProgress size={25} color="inherit" />
                        ) : (
                            'Delete'
                        )}
                    </Button>
                    <Button
                        disabled={isLoading ? true : false}
                        variant="outlined"
                        onClick={handleToggleDialogDelete}
                        color="default">
                        Cancel
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    );
};

export default EventDialog;
