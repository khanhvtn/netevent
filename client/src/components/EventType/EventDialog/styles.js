import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(() => ({
    textField: {
        marginBottom: '20px'
    },
    dialogActions: {
        marginRight: '20px'
    }
}));
