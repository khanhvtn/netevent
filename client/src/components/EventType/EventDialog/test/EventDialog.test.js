import React from 'react'
import { create } from 'react-test-renderer'
import expect from 'expect'

import EventDialog from '../EventDialog'

describe('EventDialog Component', () => {
    test('Matches the snapshot', () => {
        const eventDialog = create(<EventDialog />)
        expect(eventDialog.toJSON()).toMatchSnapshot()
    });
});