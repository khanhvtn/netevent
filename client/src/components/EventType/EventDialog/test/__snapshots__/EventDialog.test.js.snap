// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDialog Component Matches the snapshot 1`] = `<div />`;
